### ISSUE TITLE
Quartz.Net Inheritance security rules violated by type - Derived types must either match the security accessibility of the base type or be less accessible

### ISSUE THREAD

**Issue by ronnymgm** [quartznet/quartznet#258]
Hi All,

After upgrading from Quartz.Net 2.2 to version 2.3.1: 
We start receiving this exception when deploying to Windows Server 2008 R8. No problems at all in our development machines.

[[ Inheritance security rules violated by type: 'MyCompany.eCommerce.WebSiteItemHitsPurgeJob'. Derived types must either match the security accessibility of the base type or be less accessible

[TypeLoadException: Inheritance security rules violated by type: 'MyCompany.eCommerce.WebSiteItemHitsPurgeJob'. Derived types must either match the security accessibility of the base type or be less accessible.]

[JobPersistenceException: Unexpected runtime exception: Inheritance security rules violated by type: 'MyCompany.eCommerce.WebSiteItemHitsPurgeJob'. Derived types must either match the security accessibility of the base type or be less accessible.]
   System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg) +10818447
   System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type) +336
   Quartz.Simpl.IRemotableQuartzScheduler.GetJobsDetails(String jobGroup, String state, Int32 topDays, Int32 page, Int32 pageRows) +0
   Quartz.Impl.<>c__DisplayClass91.<GetJobsDetails>b__90(IRemotableQuartzScheduler x) +28
   Quartz.Impl.RemoteScheduler.CallInGuard(Func`2 func) +47
   Quartz.Impl.RemoteScheduler.GetJobsDetails(String jobGroup, String state, Int32 topDays, Int32 page, Int32 pageRows) +109]]

As you can notice our jobs inherit from JobDetailImpl, IJob and that is .. we have never experienced this type of exceptions with previous version.

By Quartz code I can see quartz contains [SecurityCritical] sections but I cannot change my code to be [SecurityCritical] instead need to follow a transparent approach. 

Maybe this is not an Quartz issue and I'm just missing something. But really need some help.

Any help will be appreciated.

**Comment by lahma** [comment]
Do you have .NET 4.x installed on the server?

**Comment by ronnymgm** [comment]
Yes I have..

Regarding this issue it was finally fixed removing the following entry in our application config file.

<runtime>
<NetFx40_LegacySecurityPolicy enabled="true" />
</runtime>

Hard to find by the way.

Thanks.

**Comment by lahma** [comment]
And were you using the .NET 3.5 Quartz dlls though? Was you app .NET 4.0 targeted? Trying to understand the combination that caused the problem.

**Comment by ronnymgm** [comment]
No 3.5 in the equation.  
My App targeted .NET 4.5. and Quartz .NET 4.0.

**Comment by gitAakash** [comment]
I am facing the issue. I am deploying my web app on **Godaddy.com**. I t works fine when deployed on other hosting environments. 
I am using .NET Version 4.6.1055 and Code access security trust level is **FULL**.
can you please help me out here..

**Comment by rx81** [comment]
gitAkash, did you ever get it working on Godaddy?  I'm in the same situation.

**Comment by fedefdx** [comment]
@gitAakash  I'm having the same problem, could you solve this issue? @rx81  what about you?

**Comment by shibin4m** [comment]
I am having the same issue while hosting into shared hosting. Any solution?

**Comment by nipek** [comment]
<trust level="Full" /> use that in the system.web

**Comment by nipek** [comment]
` <trust level="Full" />`
